### DISCUSSION TITLE
System requirements for Streamlink

### DISCUSSION THREAD

**Discussion by Colorado4Wheeler** [streamlink/streamlink#3745] [Q&A] [answered]
So far Streamlink on my main computer has been working great!  I have had 10-12 streams recording simultaneously and it's churning right along with very few errors.

When I move it to a less robust virtual computer, though, if I record more than a couple stream simultaneously they all start timing out regularly.  It's also worth noting that my main computer is online via a 400Mbps cable modem and the virtual is online via a 40Mbps DSL modem.

I'm wondering if there is a guideline for "you can record X streams if you have Y resources" or if there are additional command line options I can utilize that might help the performance of the other machine.

This also happens on a single stream.  For instance, I started recording the same stream from both computers, the virtual DSL computer got "Failed to open URL" after 14 minutes, while the main computer is still chugging along just fine (which leads me to wonder if it's just the flakiness of DSL in general).

I've tried playing with command lines a bit, but there are quite a few so it's hard to know if I'm on the right path.  Currently I use the following command line on the virtual computer (trying to compensate for DSL timeouts, etc), where the last two options are the only difference between main and virtual:

```--hls-segment-attempts 1000 --hls-segment-timeout 10 --hls-segment-attempts 10 --hls-playlist-reload-attempts 10```

Are there other command line options I should play with to try to overcome the issue or is it just going to be an ongoing issue with DSL?  I prefer the DSL connection because there's no quota so I use that for all my various big downloads and streaming.

**Comment by bastimeyer** [comment, marked as answer]
> I'm wondering if there is a guideline for "you can record X streams if you have Y resources"

There is no "generic stream type". This entirely depends on the chosen stream's bitrate, your connection to the servers and whether your system's capacity is not maxed out.

Increasing HLS segment attempts or timeouts doesn't help if your connection is simply not fast enough, and setting the value to something like 1000 is complete nonsense, as the resouces will become inaccessible after a certain amount of time on the servers in most cases.

> Are there other command line options I should play with to try to overcome the issue
> I have had 10-12 streams recording simultaneously

Don't download everything at the same time if your connection is not fast enough :facepalm:
